src/WebRequest.cpp

```diff
@@ -71,10 +71,10 @@ void AsyncWebServerRequest::_onData(void* buf, size_t len) {
 #ifndef ASYNC_TCP_SSL_ENABLED
   if (_parseState == PARSE_REQ_START && len && ((uint8_t*)buf)[0] == 0x16) { // 0x16 indicates a Handshake message (SSL/TLS).
   #ifdef ESP32
-    log_d("SSL/TLS handshake detected: closing connection");
+    log_d("SSL/TLS handshake detected: resetting connection");
   #endif
     _parseState = PARSE_REQ_FAIL;
-    _client->close();
+    _client->abort();
```

Comment:
Tries that yesterday: is dos not solve the issue and does not call the disconnect callback, so the request and client won'y be deleted. If we do that, we need to call `_onDisocnnect()` after.

Reply:
it works pretty fine for me and releases mem instantly

```
heap:263372
heap:263372
heap:263372
[ 20009][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 21235][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263340
[ 22262][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 22877][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 23288][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 23902][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
heap:263372
[ 28204][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 28612][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 29228][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
[ 32299][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 32708][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 33118][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
[ 33526][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 34959][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 35574][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 35987][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
[ 36394][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 38034][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 38649][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
[ 39059][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
[ 39465][D][WebRequest.cpp:74] _onData(): SSL/TLS handshake detected: resetting connection
heap:263372
```

Reply:
even with `autocannon -d 600 -t 600 -c 10 -M 10` it ends up with same amount of free heap before/after test

Reply:
weird!! that's what I tested and I still got the memory leak!

Reply:
let me retest with some debug logs..

Reply:
I am testing with my branch: https://github.com/mathieucarbou/ESPAsyncWebServer/tree/leak

I didn't notice that abort() was making the code trigger the onError, then onDisconnect!

```
[ 44571][D][WebRequest.cpp:215] _onError(): Client error: -13
[ 44578][D][WebRequest.cpp:231] _onDisconnect(): Client disconnected
[ 44585][D][WebServer.cpp:139] _handleDisconnect(): Deleting request pointer: 0x3ffd3cfc
[ 44594][E][WebRequest.cpp:44] operator()(): Delete client pointer: 0x3ffd4698
```

And you are right, the heap stays the same!

I think I messed up when I tested - sorry!

Reply:
@vortigont Should we then apply the same logic to reset the connection when the http request parsing fails ?

Reply:
yeah, why not. For some cases where proper connection close sequence is not required we could save resources resetting the connection!
Let's see where this could be improved!